src/tests/unit/services/GenerateVerifiableCredential.test.ts

```diff
@@ -0,0 +1,581 @@
+/* eslint-disable @typescript-eslint/dot-notation */
+import { Logger } from "@aws-lambda-powertools/logger";
+import { GenerateVerifiableCredential } from "../../../services/GenerateVerifiableCredential";
+
+describe("GenerateVerifiableCredential", () => {
+	let logger: Logger;
+	let generateVerifiableCredential: GenerateVerifiableCredential;
+
+	beforeEach(() => {
+		logger = new Logger();
+		generateVerifiableCredential = GenerateVerifiableCredential.getInstance(logger);
+	});
+
+	afterEach(() => {
+		jest.resetAllMocks();
+	});
+
+	describe("doesDocumentContainValidChip", () => {
+		it("should return true if document type is valid and contains a valid chip", () => {
+			const documentType = "PASSPORT";
+			const documentAuthenticityCheckBreakdown = [
+				{
+					sub_check: "chip_csca_trusted",
+					result: "PASS",
+				},
+			];
+
+			const result = generateVerifiableCredential["doesDocumentContainValidChip"](
+				documentType,
+				documentAuthenticityCheckBreakdown,
+			);
+
+			expect(result).toBe(true);
+		});
+
+		it("should return false if document type is not valid", () => {
+			const documentType = "INVALID_DOCUMENT";
+			const documentAuthenticityCheckBreakdown = [
+				{
+					sub_check: "chip_csca_trusted",
+					result: "PASS",
+				},
+			];
+
+			const result = generateVerifiableCredential["doesDocumentContainValidChip"](
+				documentType,
+				documentAuthenticityCheckBreakdown,
+			);
+
+			expect(result).toBe(false);
+		});
+
+		it("should return false if document type is valid but does not contain a valid chip", () => {
+			const documentType = "PASSPORT";
+			const documentAuthenticityCheckBreakdown = [
+				{
+					sub_check: "chip_csca_trusted",
+					result: "FAIL",
+				},
+			];
+
+			const result = generateVerifiableCredential["doesDocumentContainValidChip"](
+				documentType,
+				documentAuthenticityCheckBreakdown,
+			);
+
+			expect(result).toBe(false);
+		});
+	});
+
+	describe("calculateStrengthScore", () => {
+		it("should return the correct strength score for UKPASSPORT with valid chip", () => {
```

Comment:
Cant we use it.each and abstract all the variable as use-case inputs and have just 1 test case instead of duplicating the tests?

Reply:
Are we happy to do this in the future ticket planned to write Unit tests to cover all scenarios?